nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/computation/Record.java

```diff
@@ -183,14 +235,21 @@ public boolean equals(Object o) {
             return false;
         }
         Record record = (Record) o;
-        return watermark == record.watermark && Objects.equals(flags, record.flags) && Objects.equals(key, record.key)
+        return watermark == record.watermark && flagsAsByte == record.flagsAsByte && Objects.equals(key, record.key)
                 && Arrays.equals(data, record.data);
     }
 
     @Override
     public int hashCode() {
-        int result = Objects.hash(watermark, flags, key);
+        int result = Objects.hash(watermark, flagsAsByte, key);
         result = 31 * result + Arrays.hashCode(data);
         return result;
     }
+
+    public enum Flag {
+        // limited to 8 flags so it can be encoded as a byte
+        DEFAULT, COMMIT, POISON_PILL, SKIP, TRACE, PAUSE, USER1, USER2;
+
```

Comment:
what do they mean ?

Reply:
this need to be reworked see NXP-22600, for now they are not really used by the computation pattern (commit and poison_pill are still plugged but not that helpful),
their semantic belongs to upper level between producer and consumer